src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,119 @@
+import { Menu as HeadlessMenu } from '@headlessui/react';
+import clsx from 'clsx';
+import React from 'react';
+
+import styles from './Menu.module.css';
+import type { ExtractProps } from '../../util/utility-types';
+
+import ClickableStyle from '../ClickableStyle';
+import Icon from '../Icon';
+import type { IconName } from '../Icon';
+import PopoverContainer from '../PopoverContainer';
+import PopoverListItem from '../PopoverListItem';
+
+// Note: added className here to prevent private interface collision within HeadlessUI
+export type MenuProps = ExtractProps<typeof HeadlessMenu> & {
+  /**
+   * Allow custom classes to be applied to the menu container.
+   */
+  className?: string;
+};
+
+export type MenuItemProps = ExtractProps<typeof HeadlessMenu.Item> & {
+  href?: string;
+  icon?: IconName;
+  onClick?: () => any;
+};
+
+export type MenuButtonProps = ExtractProps<typeof HeadlessMenu.Button>;
+export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Menu} from "@chanzuckerberg/eds";`
+ *
+ * Creates a list of actions which appear next to a trigger component. This should
+ * be used when there is a discrete number of actions such as user navigations (e.g.,
+ * a profile menu with links to settings) or a set of actions
+ */
+export const Menu = (props: MenuProps) => {
+  return <HeadlessMenu {...props} />;
+};
+
+/**
+ * Trigger for the popover menu (dropdown)
+ */
+const MenuButton = ({ children, className, ...other }: MenuButtonProps) => {
+  const buttonClassNames = clsx(styles['menu__button'], className);
+  return (
+    <HeadlessMenu.Button
+      as={ClickableStyle}
+      className={buttonClassNames}
+      status="neutral"
+      {...other}
+    >
+      {children}
+      <Icon
+        className={styles['menu__button--icon']}
+        name="expand-more"
+        purpose="decorative"
+        size="1.5rem"
+      />
+    </HeadlessMenu.Button>
+  );
+};
+
+/**
```

Comment:
@jinlee93 are we putting subcomponents in the same file/directory now or still splitting them up?

Reply:
I actually wanted to ask you about this.
I am down to start putting subcomponents in the same file/directory as a lot of these subcomponents we really wouldn't be reusing.
However, the file does get really long, stuff could be harder to find, and css class names could get even more complex
@booc0mtaco @dierat

Reply:
My 2¢ on this one is that "bound" sub-components should live in the same folder together. Similar to how headless bundles the parts together. A rule I used in the past was "components get their own folder IFF they can be composed independent of other components".

From what I understand, BM encouraged breaking things down into separate folders, but I think that might be why some of the naming for these popover/dropdowns got wonky...? I chopped some of the reusable bits (`PopoverContainer`, `PopoverListItem`) into separate things and used the common-denominator component as a basis for the name

Reply:
>A rule I used in the past was "components get their own folder IFF they can be composed independent of other components".

This sounds right to me. I think it's confusing to have a component in its own folder if it can only be used in conjunction with another specific component—they should be bundled together. I also find it weird that there are directories with no `*.stories.ts/x` file (because subcomponents aren't exposed in storybook).

>However, the file does get really long, stuff could be harder to find, and css class names could get even more complex

I can also see splitting them up into separate files to make it cleaner, but they can still live in the same directory.

Reply:
But if we decide to move away from the one-component-per-directory rule, we should update the [code guidelines](https://chanzuckerberg.github.io/edu-design-system/iframe.html?viewMode=story&id=documentation-guidelines--code-guidelines&args=#component-rules).

Reply:
@dierat @jinlee93 yah, I don't have too strong an opinion about how we do the internals of the component, as some cases may benefit from separate subcomponent files more than others. I will turn the guidelines update into a ticket, and we can have a brief discussion at one of our upcoming meetings (or async via comments on that one). How does that sound?

Reply:
Sounds good, thanks!